### PR TITLE
spring

### PR DESCRIPTION
**Pull request by BondarenkoJek**

### CODE REVIEW

File: src/main/webapp/resources/styles/registration.css [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+.form input[type="button"]:hover,
+.form input[type="submit"]:hover {
+    background:linear-gradient(to bottom, #34CACA 5%, #30C9C9 100%);
+    background-color:#34CACA;
+}
+
+.alert-danger {
+    color: red;
+    margin-bottom: 10px;
+}
```
Comment (DmitriyLitvin) [inline src/main/webapp/resources/styles/registration.css]: empty line should be here

File: src/main/webapp/resources/styles/login.css [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+.form input[type="button"]:hover,
+.form input[type="submit"]:hover {
+    background:linear-gradient(to bottom, #34CACA 5%, #30C9C9 100%);
+    background-color:#34CACA;
+}
+
+.alert-danger {
+    color: red;
+    margin-bottom: 10px;
+}
```
Comment (DmitriyLitvin) [inline src/main/webapp/resources/styles/login.css]: empty line should be here

File: src/main/webapp/WEB-INF/pages/registration.jsp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <title>Registration</title>
+    <link href="/resources/styles/registration.css" rel="stylesheet"/>
+</head>
+<body>
+<div>
+    <form id="registrationForm" class="form" action="/registration" method="post">
+        <h2>Registration</h2>
+        <label for="userName">Login</label>
+        <input id="userName" name="userName" type="text"/>
+
```
Comment (DmitriyLitvin) [inline src/main/webapp/WEB-INF/pages/registration.jsp:13]: I ain't good at html css, but I think that empty lines between tags should have been deleted by you

File: src/main/webapp/WEB-INF/pages/login.jsp [hunk 4]
```diff
@@ -2,18 +2,23 @@
...
-    <input value="Login" type="submit">
-</form>
+        <input value="Login" type="submit">
+        <div class="div_a"><a href="/registration">registration</a></div>
+    </form>
+</div>
+<script src="/resources/js/jquery-3.3.1.js"></script>
+<script src="/resources/js/script.js"></script>
 </body>
 </html>
```
Comment (DmitriyLitvin) [inline src/main/webapp/WEB-INF/pages/login.jsp]: empty line

Comment (DmitriyLitvin) [reply src/main/webapp/WEB-INF/pages/login.jsp]: shoud be added here

File: src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -1,23 +1,29 @@
...
+        <Logger name="com.howtodoinjava" level="debug" additivity="true">
+            <appender-ref ref="fileLogger" level="debug"/>
         </Logger>
-        <Root level="INFO">
-            <AppenderRef ref="ConsoleAppender"/>
+        <Root level="debug" additivity="false">
+            <appender-ref ref="console"/>
         </Root>
     </Loggers>
 </Configuration>
```
Comment (DmitriyLitvin) [inline src/main/resources/log4j2.xml]: empty line shoud be added here

File: src/main/resources/db.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
-driver=com.mysql.jdbc.Driver
+driver=com.mysql.cj.jdbc.Driver
 url=jdbc:mysql://localhost:3306/developers
 username=root
 password=root
```
Comment (DmitriyLitvin) [inline src/main/resources/db.properties]: empty line shoud be added here

File: src/main/java/ua/bondarenkojek/util/log4j2.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    <Loggers>
+        <Logger name="ASYNC_JSON_FILE_APPENDER" level="INFO"
+                additivity="false">
+            <AppenderRef ref="AsyncAppender" />
+        </Logger>
+        <Root level="ERROR">
+            <AppenderRef ref="ConsoleAppender"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (DmitriyLitvin) [inline src/main/java/ua/bondarenkojek/util/log4j2.xml:26]: empty line shoud be added here

File: src/main/webapp/WEB-INF/pages/login.jsp (hunk 4 above)
Comment (DmitriyLitvin) [reply src/main/webapp/WEB-INF/pages/login.jsp]: Strange situation, but css styles had been usless with tag - link untlil I used construction <style><%@include file='path' %></style> where path - path to your file. So If It works - good, but check it please

**Review by DmitriyLitvin — CHANGES_REQUESTED** [review]

**Review by DmitriyLitvin — APPROVED** [review]

**Review by boroda4436 — APPROVED** [review]
LGTM

**Pull request merged**